src/Query/Geo/GeoShapeQuery.php

```diff
@@ -50,9 +55,10 @@ public function getType()
      * @param string $field       Field name.
      * @param string $type        Shape type.
      * @param array  $coordinates Shape coordinates.
+     * @param string $relation    Spatial relation.
      * @param array  $parameters  Additional parameters.
      */
-    public function addShape($field, $type, array $coordinates, array $parameters = [])
+    public function addShape($field, $type, array $coordinates, $relation = self::INTERSECTS, array $parameters = [])
```

Comment:
Dear @toooni,

IMO, this is a big change, because it could affect to the other people who is using this interface. 

Because the `shape` and `relation` are the same level in the query, what do you think if we break it out of this `addShape` function, such as `addRelation` instead.

Reply:
@trandangtri the problem is that relation is still tied to shape. But I do have an idea how to keep BC. Please wait some minutes.

Reply:
@trandangtri Now we kept BC and a deprecation notice is triggered.

Reply:
Thanks @toooni. 
CC @saimaz